### PR TITLE
Add University Service and Tests

### LINKED ISSUE #14: Add University Service and Tests
# Acceptance Criteria:

- [x] There is <tt>UniversityQueryService.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/services` directory.
- [x] <tt>UniversityQueryService.java</tt>  implements a class that make a call to the appropriate API
      and returns the results.
- [x] There is  <tt>UniversityQueryServiceTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/services` directory.
- [x] The tests in  <tt>UniversityQueryServiceTests.java</tt> provide 100% coverage of  <tt>UniversityQueryService.java</tt> for both jacoco and pitest.
      
# Details

See the assignment description at https://ucsb-cs156.github.io/f23/lab/team01 for details.

### PR DESCRIPTION
**Pull request by chelseastoner**
In this PR, we add a service that wraps the University API from http://universities.hipolabs.com/search?name={name}

Closes #14

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/UniversityQueryService.java [hunk 1]
```diff
@@ -9,24 +10,39 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.HttpClientErrorException;
 
 @Slf4j
 @Service
 public class UniversityQueryService {
 
+    ObjectMapper mapper = new ObjectMapper();
```
Comment (dayvidwang) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/UniversityQueryService.java:25]: where is mapper used?

**Review by dayvidwang — CHANGES_REQUESTED** [review]
overall looks good, I think there is an unused mapper variable (I might have this as well)

**Comment by dayvidwang** [comment]
overall terrific job

**Comment by shuyiwan** [comment]
nice work!

**Review by shuyiwan — APPROVED** [review]
Look good to me!

**Pull request merged**